easybuild/framework/easyblock.py

```diff
@@ -2150,6 +2150,22 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
             self.log.info("Loading extra modules: %s", extra_modules)
             self.modules_tool.load(extra_modules)
 
+        # Setup CUDA cache if required. If we don't do this, CUDA will use the $HOME for its cache files
+        if get_software_root('CUDA') or get_software_root('CUDAcore'):
+            cuda_cache_maxsize = build_option('cuda_cache_maxsize')
+            if cuda_cache_maxsize is None:
+                cuda_cache_maxsize = 1 * 1024 * 1024 * 1024  # 1 GB default value
+            if cuda_cache_maxsize == 0:
+                self.log.info('Disabling CUDA PTX cache as per request')
+                env.setvar('CUDA_CACHE_DISABLE', '1')
+            else:
+                cuda_cache_dir = os.path.join(self.builddir, 'eb-cuda-cache')
```

Comment:
Don't we want to make this configurable too?

I can see myself using `/dev/shm` for this for example, but always using `/dev/shm` for `--buildpath` is probably not a good idea...

Reply:
Added --cuda-cache-dir